samples/ISO-8601-2004/README.md

```diff
@@ -0,0 +1,125 @@
+# ISO 8601, Representation of dates and times
+
+Constructed by Norm Tovey-Walsh, April 2022.
+
+There are two grammars in this directory:
+
+* `iso8601.ixml` accepts an ISO 8601 date, time, datetime, duration,
+   interval, or recurrence. I believe that this grammar instantiates
+   the rules for ISO 8601:2004. It certainly doesn’t include any of
+   the new formats described in ISO 8601:2019-1,2. It also doesn’t
+   include the `--MM-DD` format(s) that were described in ISO
+   8601:2000 but not in the ISO 8601:2004. (According to
+   [Wikipedia](https://en.wikipedia.org/wiki/ISO_8601)).
+* `iso8601-list.ixml` accepts a list of ISO 8601 strings on separate lines. This version
+   only exists for testing purposes. It _should_ be identical to `iso8601.ixml` except for
+   the  additional rules `list-of-iso8601` and `comment`.
+
+The `test-data.txt` file contains a list of valid ISO 8601 strings.
+The corresponding visible XML output for those strings is in
+`test-data.xml`.
+
+## Dates and times
+
+For the most part, the output is what you’d probably
+expect. For example:
+
+```
+2022-04-17
+```
+
+becomes
+
+```xml
+<date>
+   <year>2022</year>
+   <month>04</month>
+   <day>17</day>
+</date>
+```
+
+and
+
+```
+2022-04-17T13:55:24+00:00
+```
+
+becomes
+
+```xml
+<local-datetime>
+   <year>2022</year>
+   <month>04</month>
+   <day>17</day>
+   <hour>13</hour>
+   <minute>55</minute>
+   <second>24</second>
+   <utc-offset>
+      <offset-hour>00</offset-hour>
+      <offset-minute>00</offset-minute>
+   </utc-offset>
+</local-datetime>
+```
+
+If a datetime includes a specific offset from UTC, the sign is only included on the hour if its negative.
```

Comment:
This seems like a mistake to me.
If I want to round-trip the XML back into an ISO 8601 (or similar) string, I need to have something like `if (substring( offset-hour, 1, 1 ) eq '-') then '-' else '+'`. While that is not hard, it seems unnecessary. My instinct is to put a `@direction` on `<utc-offset>` (with value `+` or `-`, because in iXML we can’t convert that to `east` or `west`).

Reply:
Fair enough. I added a note about that specifically to draw attention to the choice to see what others thought. I'm not sure I'd *want* to specify `east` or `west` to be honest, as I'm not sure I'd be able to guess which one was `+` and which one was `-` 😆

Reply:
Ok, I've pushed an update that uses `@direction` on the `offset-hour` and makes it explicitly `+` or `-` instead of making the hour value signed.

Reply:
Excellent.
Heh-heh. I was hoping for `east` and `west` precisely to make me look it up every time, so I would finally learn it!